language/bytecode-verifier/bytecode-verifier-tests/src/unit_tests/multi_pass_tests.rs

```diff
@@ -5,12 +5,13 @@ use bytecode_verifier::{
     ability_field_requirements, constants, instantiation_loops::InstantiationLoopChecker,
     DuplicationChecker, InstructionConsistency, RecursiveStructDefChecker, SignatureChecker,
 };
-use move_binary_format::CompiledModule;
+use move_binary_format::{check_bounds::BoundsChecker, CompiledModule};
 use proptest::prelude::*;
 
 proptest! {
     #[test]
     fn check_verifier_passes(module in CompiledModule::valid_strategy(20)) {
+        BoundsChecker::verify_module(&module).expect("BoundsChecker failure");
```

Comment:
This whole section feels weird. maybe just switch it to calling the main `verify_module`?

Reply:
Unfortunately I wasn't able to accomplish this easily. `CompiledModule::valid_strategy` is capable of generating modules that do not pass `friends::verify_module` verification, due to the strategy generating friend declarations that include the module itself, causing `INVALID_FRIEND_DECL_WITH_SELF` validation errors. I think the best course of action is to modify the module generator to no longer generate modules with invalid friend declarations, but I found it pretty difficult to accomplish that as part of this pull request.

I considered three possible approaches:

1. [The proptest book chapter on "Filtering"](https://altsysrq.github.io/proptest-book/proptest/tutorial/filtering.html) describes "local filtering," where the module generation strategy can filter out values with invalid friend declarations. However, while I found it simple to filter out modules with a self index of a specific value, or modules that contained a friend declaration index of a specific value, I couldn't figure out how to tie the two together: filtering out module values where the friend declaration indices included the value that the strategy chose for the self index.
2. That sounds like what [the proptest book refers to as "higher-order strategies,"](https://altsysrq.github.io/proptest-book/proptest/tutorial/higher-order.html) but I found `prop_flat_map` difficult to use, and even after several attempts, I wasn't able to filter out the invalid values. I can go into more detail on what I tried but I think it might be simpler to submit that work as a separate pull request.
3. The proptest book chapter on filtering also mentions "global filtering," in which this test could return `Err(TestCaseError::Reject)` when a module's friend declarations include itself. But that wouldn't fix the module generation strategy, and would instead just get this one test passing.

My preference would be for me to follow up on this by submitting a separate pull request to update module generation with a higher-order strategy that generated friend declarations that excluded the self index.